vms.tf

```diff
@@ -7,9 +7,11 @@ locals {
     : var.storage_type == "ha" ? module.netapp[0].netapp_endpoint : module.nfs[0].private_ip_address
   )
 
+  protocol_version = var.storage_type == "ha" && startswith(var.netapp_protocols[0], "NFS") ? split("v", var.netapp_protocols[0])[1] : "4.1"
+
   rwx_filestore_path = (var.storage_type == "none"
     ? ""
-    : var.storage_type == "ha" ? module.netapp[0].netapp_path : "/export"
+    : var.storage_type == "ha" ? module.netapp[0].netapp_path : local.protocol_version == "3" ? "/export" : "/"
```

Comment:
Again, looking to keep the directory structure intact here. The exported mount point would and should always be `/export` not `/` This would be very confusing to folks as we now actually export /export but tell them to mount /.

Reply:
The exported path is still `/export`, but because it is also the root directory NSFv4 expects you to only mount `/` not `/export`. See the `4. Note with remote NFS paths` in [NFSv4Howto](https://help.ubuntu.com/community/NFSv4Howto#NFSv4_Client)

>They don't work the way they did in NFSv3. NFSv4 has a global root directory and all exported directories are children to it. So what would have been nfs-server:/export/users on NFSv3 is nfs-server:/users on NFSv4, because /export is the root directory.

Reply:
Additional root directory is added to `/etc/exports`, see updated code in `/files/cloud-init/nfs/cloud-config`.